codegen/module_system.go

```diff
@@ -34,7 +34,11 @@ import (
 
 // NewDefaultModuleSystem creates a fresh instance of the default zanzibar
 // module system (clients, endpoints)
-func NewDefaultModuleSystem(h *PackageHelper) (*module.System, error) {
+func NewDefaultModuleSystem(
+	configDirName string,
+	middlewareConfig string,
+	h *PackageHelper,
```

Comment:
Have you considered putting the middleware config information on the PackageHelper. To me it feels a bit weird to have these arguments here. Right now I have a codebase that I only generate clients in, and the change to this interface would now require me to care about non-existent middleware configs.